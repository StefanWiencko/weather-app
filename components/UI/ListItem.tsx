import { Pressable, StyleSheet, Text, View } from "react-native";
import React from "react";
import { Colors } from "@/constants/colors";

type Props = {
  text: string;
  onPress: () => void;
};

const ListItem = ({ text, onPress }: Props) => {
  return (
    <Pressable
      android_ripple={{ color: Colors.accent200 }}
      onPress={onPress}
      style={({ pressed }) => pressed && styles.pressed}
    >
      <View style={styles.container}>
        <Text style={styles.text}>{text}</Text>
      </View>
    </Pressable>
  );
};

export default ListItem;

const styles = StyleSheet.create({
  container: {
    paddingVertical: 16,
    alignItems: "flex-start",
    borderBottomColor: Colors.grey300,
    borderBottomWidth: 1,
    backgroundColor: Colors.grey200,
  },
  text: {
    color: Colors.font,
    fontSize: 18,
    marginHorizontal: 8,
  },
  pressed: {
    opacity: 0.75,
  },
});
