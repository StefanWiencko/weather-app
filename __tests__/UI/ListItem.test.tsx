import ListItem from "@/components/UI/ListItem";
import { render, fireEvent } from "@testing-library/react-native";

describe("List item", () => {
  it("displays text correctly", () => {
    const { getByText } = render(<ListItem text="Test" onPress={() => {}} />);
    expect(getByText("Test")).toBeTruthy();
  });

  it("fires onPress fn", () => {
    const mockOnPress = jest.fn();
    const { getByText } = render(
      <ListItem text="Test" onPress={mockOnPress} />
    );

    const listItem = getByText("Test");
    fireEvent.press(listItem);

    expect(mockOnPress).toHaveBeenCalled();
  });

  it("matches snapshot", () => {
    const tree = render(<ListItem text="Test" onPress={() => {}} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
