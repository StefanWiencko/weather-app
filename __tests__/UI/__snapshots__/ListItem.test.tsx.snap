// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List item matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={false}
>
  <View
    style={
      {
        "alignItems": "flex-start",
        "backgroundColor": "#0000000d",
        "borderBottomColor": "#00000033",
        "borderBottomWidth": 1,
        "paddingVertical": 16,
      }
    }
  >
    <Text
      style={
        {
          "color": "#222",
          "fontSize": 18,
          "marginHorizontal": 8,
        }
      }
    >
      Test
    </Text>
  </View>
</View>
`;
